### DISCUSSION TITLE
How to add resolve option for domains containing dashes

### DISCUSSION THREAD

**Discussion by michel-zimmer** [Orange-OpenSource/hurl#2829] [Q&A] [answered]
Hi everyone,

this works as expected:

```hurl
GET https://foo.bar.com/
[Options]
resolve: foo.bar.com:443:192.168.0.110
HTTP 200
```

But i cannot the following to work:

```hurl
GET https://foo-bar.com/
[Options]
resolve: foo-bar.com:443:192.168.0.110
HTTP 200
```

```
error: Parsing literal
  --> example.hurl:3:10
   |
 3 | resolve: foo-bar.com:443:192.168.0.110
   |          ^ expecting 'HOST:PORT:ADDR'
   |
```

No matter how I try to escape the minus sign or how I try to quote parts of the line or the entire line, I cannot get the redirect option to work with domains containing minus signs. Same for the command line parameter.

Does someone have an idea, what the problem might be?
Thanks and kind regards :)

**Comment by jcamiel** [comment]
Hi @michel-zimmer 
The Hurl file syntax looks OK to me, so it maybe a runtime error.
Could you paste here the exact command line you run and the result of running it with `--verbose`.
In these logs file, there should be the equivalent curl command, could you try ti run this curl command also ?

**Reply by michel-zimmer** [reply, marked as answer]
Hi @jcamiel 

```shell
$ hurl --verbose example.hurl
error: Parsing literal
  --> example.hurl:3:10
   |
 3 | resolve: foo-bar.com:443:192.168.0.110
   |          ^ expecting 'HOST:PORT:ADDR'
   |
```

```shell
$ hurl --version 
hurl 4.1.0 (unknown) libcurl/8.3.0-DEV OpenSSL/3.0.13 zlib/1.3
Features (libcurl):  alt-svc AsynchDNS HSTS IPv6 libz SSL UnixSockets
Features (built-in): brotli
```

```shell
$ cat example.hurl
GET https://foo-bar.com/
[Options]
resolve: foo-bar.com:443:192.168.0.110
HTTP 200
```

Displaying the version gave me the idea to look up the most recent version and I was actually behind.
I'm using NixOS and stable (23.11) is on 4.1.0 while unstable already has 4.3.0.
And using 4.3.0 works as expected.

```shell
$ hurl --verbose example.hurl
* ------------------------------------------------------------------------------
* Executing entry 1
*
* Entry options:
* resolve: foo-bar.com:443:192.168.0.110
*
* Cookie store:
*
* Request:
* GET https://foo-bar.com/
*
* Request can be run with the following curl command:
* curl --resolve foo-bar.com:443:192.168.0.110 'https://foo-bar.com/'
*
error: HTTP connection
  --> example.hurl:1:5
   |
 1 | GET https://foo-bar.com/
   |     ^^^^^^^^^^^^^^^^^^^^ (7) Failed to connect to foo-bar.com port 443 after 3102 ms: Couldn't connect to server
   |
```

```shell
$ hurl --version
hurl 4.3.0 (unknown) libcurl/8.6.0-DEV OpenSSL/3.0.13 zlib/1.3.1
Features (libcurl):  alt-svc AsynchDNS HSTS IPv6 libz SSL UnixSockets
Features (built-in): brotli
```

From the error message of the 4.1.0 version I would guess there was a parsing error and that it is already fixed in 4.3.0.

So thanks a lot :)

**Reply by jcamiel** [reply]
Perfect, I should have asked you the version first!
